extern/magic_enum/magic_enum.hpp

```diff
@@ -0,0 +1,1131 @@
+//  __  __             _        ______                          _____
```

Comment:
Can we include this by submodule rather than copying it into our project?

Reply:
Changed to submodule